buildSrc/subprojects/parser/src/test/java/org/apache/calcite/test/KeywordTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.test;
+
+import org.apache.calcite.buildtools.parser.Keyword;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Assertions;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class KeywordTest {
+
+  @Test public void testHashCodeEquality() {
+    Keyword a = new Keyword("foo");
+    Keyword b = new Keyword("foo", "path/file");
+    assertEquals(a.hashCode(), b.hashCode());
+    assertEquals(a, a);
```

Comment:
Why is this assertion here? It doesn't seem to go along with the name of the test.

Reply:
Good catch, I meant to put that in a different test.